### ISSUE TITLE
Can't Setup Plugins

### ISSUE THREAD

**Issue by bl0124** [hoobs-org/HOOBS#1551]
**Description**
Just got HOOBS in a box. Wired connection into my network and created username and password. Installed Ring plugin and I see the Ring Bridge.  But I don't see any QR codes to add into Homekit.  Also I don't see where I could log into my Ring account to get accessories to show up or where I can put in a Refresh Token like some of the YouTube videos demonstrated.  I tried other plugins and no QR codes show up for those either.

**Version**
HOOBS in a Box, Version: 4.0.104

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [X] Yes
- [ ] No

**Pervious version**
I don't know, the box is new to me, I powered it up, created log-in, saw there was an update available and I clicked on Update.

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [ ] Yes
- [X] No, I don't know

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [X] HOOBS Box
- [ ] HOOBS SD Card
- [ ] Custom Install

**If this is a custom install, what's your operating system?**
n/a

**List your plugins**
Ring

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
{
    "accessories": [],
    "platforms": [
        {
            "platform": "Ring"
        }
    ]
}
```

**Post your log**
You can get the log from the interface. This is in the Log section.

```
/28/2021, 10:06:53 PMmyqbridge stopping
9/28/2021, 10:06:58 PMmyQ Bridge starting
9/28/2021, 10:07:40 PMringbridge stopping
9/28/2021, 10:07:45 PMRing Bridge starting
9/28/2021, 10:12:38 PMringbridge stopping
9/28/2021, 10:12:43 PMRing Bridge starting
9/28/2021, 10:13:16 PMWARNINGdevice shutdown command received
9/28/2021, 10:13:17 PMRing Bridge stopping
9/28/2021, 10:13:17 PMmyQ Bridge stopping
9/28/2021, 10:13:41 PMhub is running on port 80
9/28/2021, 10:13:42 PMRing Bridge starting
9/28/2021, 10:13:42 PMmyQ Bridge starting
9/28/2021, 10:20:01 PMringbridge stopping
9/28/2021, 10:20:06 PMRing Bridge starting
9/28/2021, 10:26:15 PMringbridge stopping
9/28/2021, 10:26:20 PMRing Bridge starting
9/28/2021, 10:32:28 PMRing Bridge 2 starting
9/28/2021, 10:33:15 PMringbridge2 stopping
9/28/2021, 10:33:21 PMRing Bridge 2 starting
9/28/2021, 10:34:11 PMringbridge2 stopping
9/28/2021, 10:34:16 PMRing Bridge 2 starting
9/28/2021, 10:36:28 PMringbridge stopping
9/28/2021, 10:36:33 PMRing Bridge starting
9/28/2021, 10:50:28 PMringbridge2 stopping
9/28/2021, 10:50:35 PMmyqbridge stopping
9/28/2021, 10:50:47 PMringbridge stopping
9/28/2021, 10:50:52 PMRing Bridge starting
9/28/2021, 10:51:05 PMringbridge stopping
9/28/2021, 10:51:10 PMRing Bridge starting
9/28/2021, 10:51:11 PMringbridge stopping
9/28/2021, 10:51:14 PMRing Bridge starting
9/28/2021, 10:52:06 PMringbridge stopping
9/28/2021, 10:52:12 PMRing Bridge starting
9/28/2021, 10:54:20 PMHue Bridge starting
9/28/2021, 10:55:21 PMhuebridge stopping
9/28/2021, 10:55:26 PMHue Bridge starting
9/28/2021, 11:04:19 PMhuebridge stopping
```

**Additional context**
This has not been a user friendly experience.

**Comment by askovi** [comment]
The QR code in HOOBS 4 is on the bridge page. Each bridge has its own QR code. The bridge must be up and running for it to show. You have to configure Ring from the plugin configuration page. Click the bottom left cog icon then click on Ring and click on Link Account and follow the onscreen Ring login prompt.

**Comment by bl0124** [comment]
When I go to the Ring Bridge, no QR code is shown.  Under Pairing, i see either Start or Cache.  I clicked Start, it thinks for a minute, then back to the Ring Bridge page with no changes, no QR.  

Went to the plugin config page by click button left cog, clicked on Ring, I see "Ring Bridge" tab with a bunch of option boxes, but nothing for Link Account.

**Comment by askovi** [comment]
The bridge must be started in order for the QR code to appear and for the Link Account button to show in the configuration. If it's not starting then there must be another issue with it. Click the Start button then head over to the logs page from the left menu and copy and paste here what you're getting.

**Comment by bl0124** [comment]
I clicked "Start" in the Ring Bridge page under Pairing (I assume that's what you meant).  Then the page went 'thinking' for about 20-30 seconds, then it came back to the Ring Bridge page with no QR code (and no Link Account in the plugin config page).  Here is the log file from when I clicked the Start:

9/29/2021, 8:49:38 AMringbridge stopping
9/29/2021, 8:49:44 AMRing Bridge starting

**Comment by askovi** [comment]
Can you please delete all your ring related bridges because the logs show more than one. Start fresh by installing the ring plugin onto a new bridge from the plugins page. Don't create the bridge manually, it will automatically get created when you install the plugin

**Comment by bl0124** [comment]
Yes, I deleted all of the other bridges last night, and I only have one Ring Bridge and no other bridges running this morning at 8:49am when I clicked Start.

**Comment by bl0124** [comment]
Yes, i've deleted all other bridges.  After your reply, I just deleted the Ring Bridge to where I have no bridges at all, then reinstalled the Ring plugin.  I only have the Ring Bridge, no other bridges.  Still same symptoms.  Here is the log from this morning onwards including my uninstall and re-install of Ring:

9/29/2021, 8:49:38 AMringbridge stopping
9/29/2021, 8:49:44 AMRing Bridge starting
9/29/2021, 9:47:00 AMringbridge stopping
9/29/2021, 9:47:40 AMRing Bridge starting
9/29/2021, 9:48:25 AMringbridge stopping
9/29/2021, 9:48:30 AMRing Bridge starting
9/29/2021, 9:48:44 AMringbridge stopping
9/29/2021, 9:48:49 AMRing Bridge starting

**Comment by askovi** [comment]
If you click the little debug icon at the top of the logs page, do you see additional logs?

**Comment by bl0124** [comment]
Yes, here's what it shows, very long, hope it all shows up here:

9/29/2021, 8:49:38 AMringbridge stopping
9/29/2021, 8:49:44 AMRing Bridge starting
9/29/2021, 9:46:51 AMyarn remove v1.22.11
9/29/2021, 9:46:51 AM[1/3] Removing module homebridge-ring...
9/29/2021, 9:46:51 AM[2/3] Removing module hoobs-ring-sidecar...
9/29/2021, 9:46:52 AM[3/3] Regenerating lockfile and installing missing dependencies...
9/29/2021, 9:46:59 AMsuccess Uninstalled packages.
9/29/2021, 9:46:59 AMDone in 8.93s.
9/29/2021, 9:47:00 AMringbridge stopping
9/29/2021, 9:47:40 AMRing Bridge starting
9/29/2021, 9:47:52 AMyarn add v1.22.11
9/29/2021, 9:47:52 AMinfo No lockfile found.
9/29/2021, 9:47:52 AM[1/4] Resolving packages...
9/29/2021, 9:47:59 AM[2/4] Fetching packages...
9/29/2021, 9:48:02 AM[3/4] Linking dependencies...
9/29/2021, 9:48:15 AM[4/4] Building fresh packages...
9/29/2021, 9:48:23 AMsuccess Saved lockfile.
9/29/2021, 9:48:23 AMsuccess Saved 148 new dependencies.
9/29/2021, 9:48:23 AMinfo Direct dependencies
9/29/2021, 9:48:23 AM├─ homebridge-ring@9.21.2
9/29/2021, 9:48:23 AM└─ hoobs-ring-sidecar@1.0.9
9/29/2021, 9:48:23 AMinfo All dependencies
9/29/2021, 9:48:23 AM├─ @homebridge/camera-utils@2.0.4
9/29/2021, 9:48:23 AM├─ @homebridge/plugin-ui-utils@0.0.19
9/29/2021, 9:48:23 AM├─ @sindresorhus/is@4.2.0
9/29/2021, 9:48:23 AM├─ @szmarczak/http-timer@4.0.6
9/29/2021, 9:48:23 AM├─ @types/cacheable-request@6.0.2
9/29/2021, 9:48:23 AM├─ @types/http-cache-semantics@4.0.1
9/29/2021, 9:48:23 AM├─ @types/keyv@3.1.3
9/29/2021, 9:48:23 AM├─ @types/responselike@1.0.0
9/29/2021, 9:48:23 AM├─ @types/socket.io-client@1.4.36
9/29/2021, 9:48:23 AM├─ after@0.8.2
9/29/2021, 9:48:23 AM├─ array-find-index@1.0.2
9/29/2021, 9:48:23 AM├─ arraybuffer.slice@0.0.7
9/29/2021, 9:48:23 AM├─ arrify@1.0.1
9/29/2021, 9:48:23 AM├─ axios@0.21.4
9/29/2021, 9:48:23 AM├─ backo2@1.0.2
9/29/2021, 9:48:23 AM├─ base64-arraybuffer@0.1.4
9/29/2021, 9:48:23 AM├─ binary-data@0.6.0
9/29/2021, 9:48:23 AM├─ blob@0.0.5
9/29/2021, 9:48:23 AM├─ buffer-xor@2.0.2
9/29/2021, 9:48:23 AM├─ cacheable-lookup@5.0.4
9/29/2021, 9:48:23 AM├─ cacheable-request@7.0.2
9/29/2021, 9:48:23 AM├─ call-bind@1.0.2
9/29/2021, 9:48:23 AM├─ camelcase-keys@4.2.0
9/29/2021, 9:48:23 AM├─ chownr@2.0.0
9/29/2021, 9:48:23 AM├─ clone-response@1.0.2
9/29/2021, 9:48:23 AM├─ colors@1.4.0
9/29/2021, 9:48:23 AM├─ component-bind@1.0.0
9/29/2021, 9:48:23 AM├─ component-inherit@0.0.3
9/29/2021, 9:48:23 AM├─ cross-spawn@7.0.3
9/29/2021, 9:48:23 AM├─ currently-unhandled@0.4.1
9/29/2021, 9:48:23 AM├─ decamelize-keys@1.1.0
9/29/2021, 9:48:23 AM├─ decamelize@1.2.0
9/29/2021, 9:48:23 AM├─ decode-uri-component@0.2.0
9/29/2021, 9:48:23 AM├─ decompress-response@6.0.0
9/29/2021, 9:48:23 AM├─ defer-to-connect@2.0.1
9/29/2021, 9:48:23 AM├─ detect-libc@1.0.3
9/29/2021, 9:48:23 AM├─ dotenv@8.6.0
9/29/2021, 9:48:23 AM├─ end-of-stream@1.4.4
9/29/2021, 9:48:23 AM├─ engine.io-client@3.5.2
9/29/2021, 9:48:23 AM├─ engine.io-parser@2.2.1
9/29/2021, 9:48:23 AM├─ error-ex@1.3.2
9/29/2021, 9:48:23 AM├─ execa@5.1.1
9/29/2021, 9:48:23 AM├─ ffmpeg-for-homebridge@0.0.9
9/29/2021, 9:48:23 AM├─ filter-obj@1.1.0
9/29/2021, 9:48:23 AM├─ find-up@2.1.0
9/29/2021, 9:48:23 AM├─ follow-redirects@1.14.4
9/29/2021, 9:48:23 AM├─ fs-minipass@2.1.0
9/29/2021, 9:48:23 AM├─ generate-function@2.3.1
9/29/2021, 9:48:23 AM├─ get-stream@5.2.0
9/29/2021, 9:48:23 AM├─ got@11.8.2
9/29/2021, 9:48:23 AM├─ graceful-fs@4.2.8
9/29/2021, 9:48:23 AM├─ has-cors@1.1.0
9/29/2021, 9:48:23 AM├─ has-symbols@1.0.2
9/29/2021, 9:48:23 AM├─ homebridge-ring@9.21.2
9/29/2021, 9:48:23 AM├─ hoobs-ring-sidecar@1.0.9
9/29/2021, 9:48:23 AM├─ hosted-git-info@2.8.9
9/29/2021, 9:48:23 AM├─ http-cache-semantics@4.1.0
9/29/2021, 9:48:23 AM├─ http2-wrapper@1.0.3
9/29/2021, 9:48:23 AM├─ human-signals@2.1.0
9/29/2021, 9:48:23 AM├─ indent-string@3.2.0
9/29/2021, 9:48:23 AM├─ ip@1.1.5
9/29/2021, 9:48:23 AM├─ ip2buf@2.0.0
9/29/2021, 9:48:23 AM├─ is-arrayish@0.2.1
9/29/2021, 9:48:23 AM├─ is-core-module@2.7.0
9/29/2021, 9:48:23 AM├─ is-plain-obj@1.1.0
9/29/2021, 9:48:23 AM├─ is-plain-object@2.0.4
9/29/2021, 9:48:23 AM├─ is-property@1.0.2
9/29/2021, 9:48:23 AM├─ is-stream@2.0.1
9/29/2021, 9:48:23 AM├─ is-stun@2.0.0
9/29/2021, 9:48:23 AM├─ isexe@2.0.0
9/29/2021, 9:48:23 AM├─ isobject@3.0.1
9/29/2021, 9:48:23 AM├─ json-buffer@3.0.1
9/29/2021, 9:48:23 AM├─ json-parse-better-errors@1.0.2
9/29/2021, 9:48:23 AM├─ keyv@4.0.3
9/29/2021, 9:48:23 AM├─ load-json-file@4.0.0
9/29/2021, 9:48:23 AM├─ locate-path@2.0.0
9/29/2021, 9:48:23 AM├─ loud-rejection@1.6.0
9/29/2021, 9:48:23 AM├─ map-obj@1.0.1
9/29/2021, 9:48:23 AM├─ meow@5.0.0
9/29/2021, 9:48:23 AM├─ merge-stream@2.0.0
9/29/2021, 9:48:23 AM├─ mimic-fn@2.1.0
9/29/2021, 9:48:23 AM├─ mimic-response@3.1.0
9/29/2021, 9:48:23 AM├─ minimist-options@3.0.2
9/29/2021, 9:48:23 AM├─ minizlib@2.1.2
9/29/2021, 9:48:23 AM├─ ms@2.0.0
9/29/2021, 9:48:23 AM├─ normalize-package-data@2.5.0
9/29/2021, 9:48:23 AM├─ normalize-url@4.5.1
9/29/2021, 9:48:23 AM├─ npm-run-path@4.0.1
9/29/2021, 9:48:23 AM├─ object-inspect@1.11.0
9/29/2021, 9:48:23 AM├─ once@1.4.0
9/29/2021, 9:48:23 AM├─ onetime@5.1.2
9/29/2021, 9:48:23 AM├─ p-cancelable@2.1.1
9/29/2021, 9:48:23 AM├─ p-limit@1.3.0
9/29/2021, 9:48:23 AM├─ p-locate@2.0.0
9/29/2021, 9:48:23 AM├─ p-try@1.0.0
9/29/2021, 9:48:23 AM├─ parse-json@4.0.0
9/29/2021, 9:48:23 AM├─ parse-path@4.0.3
9/29/2021, 9:48:23 AM├─ parse-url@5.0.7
9/29/2021, 9:48:23 AM├─ path-exists@3.0.0
9/29/2021, 9:48:23 AM├─ path-key@3.1.1
9/29/2021, 9:48:23 AM├─ path-parse@1.0.7
9/29/2021, 9:48:23 AM├─ path-type@3.0.0
9/29/2021, 9:48:23 AM├─ pick-port@1.0.0
9/29/2021, 9:48:23 AM├─ protocols@1.4.8
9/29/2021, 9:48:23 AM├─ pump@3.0.0
9/29/2021, 9:48:23 AM├─ qs@6.10.1
9/29/2021, 9:48:23 AM├─ query-string@6.14.1
9/29/2021, 9:48:23 AM├─ quick-lru@5.1.1
9/29/2021, 9:48:23 AM├─ read-pkg-up@3.0.0
9/29/2021, 9:48:23 AM├─ read-pkg@3.0.0
9/29/2021, 9:48:23 AM├─ redent@2.0.0
9/29/2021, 9:48:23 AM├─ resolve-alpn@1.2.1
9/29/2021, 9:48:23 AM├─ resolve@1.20.0
9/29/2021, 9:48:23 AM├─ rxjs@7.3.0
9/29/2021, 9:48:23 AM├─ safe-buffer@5.2.1
9/29/2021, 9:48:23 AM├─ sdp@3.0.3
9/29/2021, 9:48:23 AM├─ semver@5.7.1
9/29/2021, 9:48:23 AM├─ shebang-command@2.0.0
9/29/2021, 9:48:23 AM├─ shebang-regex@3.0.0
9/29/2021, 9:48:23 AM├─ side-channel@1.0.4
9/29/2021, 9:48:23 AM├─ signal-exit@3.0.4
9/29/2021, 9:48:23 AM├─ simple-concat@1.0.1
9/29/2021, 9:48:23 AM├─ simple-get@3.1.0
9/29/2021, 9:48:23 AM├─ socket.io-client@2.4.0
9/29/2021, 9:48:23 AM├─ socket.io-parser@3.3.2
9/29/2021, 9:48:23 AM├─ spdx-correct@3.1.1
9/29/2021, 9:48:23 AM├─ spdx-exceptions@2.3.0
9/29/2021, 9:48:23 AM├─ split-on-first@1.1.0
9/29/2021, 9:48:23 AM├─ strict-uri-encode@2.0.0
9/29/2021, 9:48:23 AM├─ strip-bom@3.0.0
9/29/2021, 9:48:23 AM├─ strip-final-newline@2.0.0
9/29/2021, 9:48:23 AM├─ strip-indent@2.0.0
9/29/2021, 9:48:23 AM├─ stun@2.1.0
9/29/2021, 9:48:23 AM├─ systeminformation@5.8.7
9/29/2021, 9:48:23 AM├─ tar@6.1.11
9/29/2021, 9:48:23 AM├─ to-array@0.1.4
9/29/2021, 9:48:23 AM├─ trim-newlines@2.0.0
9/29/2021, 9:48:23 AM├─ tslib@2.1.0
9/29/2021, 9:48:23 AM├─ turbo-crc32@1.0.1
9/29/2021, 9:48:23 AM├─ universalify@0.1.2
9/29/2021, 9:48:23 AM├─ uuid@8.3.2
9/29/2021, 9:48:23 AM├─ validate-npm-package-license@3.0.4
9/29/2021, 9:48:23 AM├─ which@2.0.2
9/29/2021, 9:48:23 AM├─ wrappy@1.0.2
9/29/2021, 9:48:23 AM├─ ws@7.4.6
9/29/2021, 9:48:23 AM├─ xmlhttprequest-ssl@1.6.3
9/29/2021, 9:48:23 AM├─ yargs-parser@10.1.0
9/29/2021, 9:48:23 AM└─ yeast@0.1.2
9/29/2021, 9:48:23 AMDone in 31.50s.
9/29/2021, 9:48:25 AMringbridge stopping
9/29/2021, 9:48:30 AMRing Bridge starting
9/29/2021, 9:48:44 AMringbridge stopping
9/29/2021, 9:48:49 AMRing Bridge starting

**Comment by bl0124** [comment]
I've attached some screenshots of what I see to illustrate I don't see Link Account and I don't see QR code

![Screen Shot 2021-09-29 at 14 18 23](https://user-images.githubusercontent.com/91588398/135350058-4eef11c4-1371-48c3-963d-3c08c957cde1.png)
![Screen Shot 2021-09-29 at 14 18 41](https://user-images.githubusercontent.com/91588398/135350100-79bf6630-bd31-4e6e-a9d4-835ac1fd28b0.png)
![Screen Shot 2021-09-29 at 14 18 53](https://user-images.githubusercontent.com/91588398/135350120-6177ce91-d091-49b7-b1b2-a9f4c62edd10.png)

**Comment by bl0124** [comment]
... and log file:

[log.txt](https://github.com/hoobs-org/HOOBS/files/7255645/log.txt)

**Comment by mkellsy** [comment]
There was an error in the update check. You will need to upgrade from the terminal.
Run these 3 commands
```
sudo apt update && sudo apt install -y hoobs-cli hoobsd && sudo systemctl restart hoobsd
```

**Comment by bl0124** [comment]
Michael, that *almost* fixed it.  It didn't fix the QR and Link Account issue right away; I still didn't see QR and Link Account.  I deleted the plug-in, rebooted, and reinstalled, still nothing.  I just so happened to decide to click "Check Update" on the dashboard to see if there was yet another update after running those commands you gave, and a GUI update was available.  I updated, and then went back to the plugin config page and voila!  Link Account button was there.  I entered username/pw and 2fa, then the bridge started and I see the QR code, and accessories populated.  I checked the log file, and it is a lot more colorful than what I've seen up until now (shows yellow and red now).

So what was the actual issue?  I think others are running into the same issue (I saw a similar post earlier today).

**Comment by mkellsy** [comment]
The cause was Node’s WS engine. It wasn’t closing connections and was saturating the available connections. I changed to EIOWS which is a much cleaner and faster WS engine.

**Comment by bl0124** [comment]
thanks @mkellsy for the help.  i hope it is a fix for everything else too.
